crawl-ref/source/webserver/game_data/static/cell_renderer.js

```diff
@@ -111,6 +111,59 @@ function ($, view_data, main, tileinfo_player, icons, dngn, enums,
                 + "px " + this.glyph_mode_font);
         },
 
+        glyph_mode_update_font_metrics: function ()
+        {
+            this.ctx.font = this.glyph_mode_font_name();
+
+            // Glyph used here does not matter because fontBoundingBoxAscent
+            // and fontBoundingBoxDescent are specific to the font whereas all
+            // glyphs in a monospaced font will have the same width
+            var metrics = this.ctx.measureText('@');
+            this.glyph_mode_font_width = metrics.width;
+
+            // Currently, fontBoundingBoxAscent/Descent are still
+            // experimental for most web browsers and may be unavailable.
+            if (metrics.fontBoundingBoxAscent)
+            {
+                this.glyph_mode_baseline = metrics.fontBoundingBoxAscent;
+                this.glyph_mode_line_height = metrics.fontBoundingBoxAscent
+                                            + metrics.fontBoundingBoxDescent;
+            }
+            else
+            {   // Inspired by https://stackoverflow.com/q/1134586/
+                var body = document.body;
+                var dmy_glyph = document.createElement("span");
```

Comment:
I'm confused by this variable name, this span element is part of what you use to determine the height.